tests/admin/test-class-admin-recommended-replace-vars.php

```diff
@@ -0,0 +1,227 @@
+
+		update_option( 'show_on_front', $show_on_front );
+		update_option( 'page_on_front', $page_on_front );
+	}
+
+	/**
+	 * Test that determine_for_post can detect a page.
+	 *
+	 * @covers WPSEO_Admin_Recommended_Replace_Vars::determine_for_post
+	 */
+	public function test_determine_for_post_with_a_page() {
+		$post = $this->create_and_get_with_post_type( 'page' );
+
+		$this->assertEquals( 'page', $this->class_instance->determine_for_post( $post ) );
+	}
+
+	/**
+	 * Test that determine_for_post can detect a post.
+	 *
+	 * @covers WPSEO_Admin_Recommended_Replace_Vars::determine_for_post
+	 */
+	public function test_determine_for_post_with_a_post() {
+		$post = $this->create_and_get_with_post_type( 'post' );
+
+		$this->assertEquals( 'post', $this->class_instance->determine_for_post( $post ) );
+	}
+
+	/**
+	 * Test that determine_for_post can detect a WooCommerce product.
+	 *
+	 * @covers WPSEO_Admin_Recommended_Replace_Vars::is_woocommerce_active
+	 * @covers WPSEO_Admin_Recommended_Replace_Vars::determine_for_post
+	 */
+	public function test_determine_for_post_with_a_woocommerce_product() {
+		$class_instance = $this->mock_is_woocommerce_active( true );
+		$post = $this->create_and_get_with_post_type( 'product' );
+
+		$this->assertEquals( 'product', $class_instance->determine_for_post( $post ) );
+	}
+
+	/**
+	 * Test that determine_for_post can not detect a WooCommerce product.
+	 *
+	 * @covers WPSEO_Admin_Recommended_Replace_Vars::is_woocommerce_active
+	 * @covers WPSEO_Admin_Recommended_Replace_Vars::determine_for_post
+	 */
+	public function test_determine_for_post_with_a_woocommerce_product_without_woocommerce() {
+		$class_instance = $this->mock_is_woocommerce_active( false );
+		$post = $this->create_and_get_with_post_type( 'product' );
+
+		$this->assertEquals( 'custom_post_type', $class_instance->determine_for_post( $post ) );
+	}
+
+	/**
+	 * Test that determine_for_post can detect a custom post type.
+	 *
+	 * @covers WPSEO_Admin_Recommended_Replace_Vars::determine_for_post
+	 */
+	public function test_determine_for_post_with_a_custom_post_type() {
+		$post = $this->create_and_get_with_post_type( 'some_plugin_post' );
+
+		$this->assertEquals( 'custom_post_type', $this->class_instance->determine_for_post( $post ) );
+	}
+
+	/**
+	 * Create and get a mocked WP_Post with a certain post_type.
+	 *
+	 * @param string $post_type The post type to give to the post.
+	 *
+	 * @return WP_Post A mocked post with the specified post_type.
+	 */
+	private function create_and_get_with_post_type( $post_type = 'post' ) {
+		return self::factory()->post->create_and_get(
+			array(
+				'post_type' => $post_type
+			)
+		);
+	}
+
+	/**
+	 * Mock helper: WPSEO_Admin_Recommended_Replace_Vars mocked for is_woocommerce_active.
+	 *
+	 * @param bool $is_active Whether WooCommerce is active or not
+	 *
+	 * @return WPSEO_Admin_Recommended_Replace_Vars A mocked
+	 *                                              WPSEO_Admin_Recommended_Replace_Vars
```

Comment:
This line is weird. I would change this documentation to: The mocked WPSEO_Admin_Recommended_Replace_Vars object.

I a developer wants to know what it exactly does he can look into the code. Also: if the implementation will change you shouldn't have to change the docs.